ompi/mca/mtl/psm2/mtl_psm2_component.c

```diff
@@ -323,6 +323,18 @@ ompi_mtl_psm2_component_open(void)
 static int
 ompi_mtl_psm2_component_query(mca_base_module_t **module, int *priority)
 {
+
+    /*
+     * Mixing the PSM2 MTL with the OFI BTL (using PSM2) 
+     * can cause an issue when they both call psm2_finalize
+     * in older versions of libpsm2.
+     */
+    if (!psm2_get_capability_mask(PSM2_LIB_REFCOUNT_CAP)) {
```

Comment:
looks like this symbol isn't defined for our Intel HFI1 clusters psm2 install.  Need to protect this code with ifdef's.

Reply:
> looks like this symbol isn't defined for our Intel HFI1 clusters psm2 install. Need to protect this code with ifdef's.

Howard, that function has been in libpsm2 for 3 years now. If your cluster is running a PSM2 that old it can't run the OFI MTL properly and has other problems as well.

Reply:
Okay, so now I'm not sure what to do - if OMPI 5.x is built with a version of PSM2 that doesn't have PSM2_LIB_REFCOUNT_CAP defined, the PSM2 MTL isn't going to operate correctly.

The whole point of this patch is to require the correct version of libpsm2...

Reply:
I mean, I can add a #warning or #error...

Reply:
Which version of PSM2 defines PSM2_LIB_REFCOUNT_CAP?
We'd rather have these kind of restrictions enforced in the configury stage, so if one configured OMPI with --with-psm2 and the psm2 lib installed on the system is too old, the configure will fail with a hopefully informative error message.

Reply:
I've pushed a patch that #defines the constant if it isn't defined. This should allow compilation to complete with the older libpsm2, but it will not execute until the libpsm2 is updated.

Reply:
> We'd rather have these kind of restrictions enforced in the configury stage,

Sorry, I didn't see your reply before I pushed the previous commit.

I would agree but I know nothing about how to modify m4 files. This might take a while...

Reply:
If the version of PSM2 is too old, configure will how report:

> PSM2 needs to be version 11.2.173 or later. Disabling MTL.

This is comparable to the configure error that occurs if you try to enable progress threads with PSM2.